addons/binding/org.openhab.binding.silvercrestwifisocket/src/main/java/org/openhab/binding/silvercrestwifisocket/internal/enums/SilvercrestWifiSocketVendor.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.silvercrestwifisocket.internal.enums;
+
+/**
+ * This enum represents the available Wifi Socket vendors.
+ *
+ * @author Christian Heimerl - Initial contribution
+ *
+ */
+public enum SilvercrestWifiSocketVendor {
+    Lidl_Silvercrest("C1", "7150"),
+    Aldi_EasyHome("C2", "92DD");
+
+    private String companyCode;
+    private String authenticationCode;
+
+    private SilvercrestWifiSocketVendor(final String companyCode, final String authenticationCode) {

```

Comment:
It's private be default you can leave it out and please leave out final on local variables and parameters in (new) code.

Reply:
Just out of curiosity: Why make fields final if possible (you're right with the comment above) but not parameters or local variables?

Reply:
For local variables the scope is very limited (or should be in the case you make your methods to big) and the same holds for parameters but there we have a check in our standard IDE configuration.

However for the fields the scope can be way bigger and it really helps to know whether the field will / can be changed.